crates/edr_solidity/src/contracts_identifier/radix_tree.rs

```diff
@@ -0,0 +1,540 @@
+use std::collections::BTreeMap;
+
+use revm_primitives::Bytes;
+
+#[derive(Debug)]
+pub struct RadixNode {
+    content: Bytes,
+    is_present: bool,
+    bytes_matched_before: usize,
```

Comment:
Is this an optimisation? It seems redundant as you can accumulate the content length (already pre-calculated) as you traverse the tree

Reply:
It's an optimization, yes. I tried to keep the original approach as much as possible to get this done faster, so I'm not 100% sure if it's a valid optimization or not.